source/defclass-star.lisp

```diff
@@ -253,29 +265,40 @@ Use the slot name directly:
         (*symbols-to-export* nil)
         (*export-class-name-p* export-class-name)
         (*export-accessor-names-p* export-accessor-names)
-        (*export-slot-names-p* export-slot-names))
+        (*export-slot-names-p* export-slot-names)
+        (*export-predicate-name-p* export-predicate-name)
+        (*predicate-name-transformer* predicate-name-transformer))
     (multiple-value-bind (binding-names binding-values clean-options)
         (extract-options-into-bindings options)
       (progv binding-names (mapcar #'eval binding-values)
         (let ((result (funcall expansion-builder
                                (mapcar 'process-slot-definition slots)
                                clean-options)))
-          (if (or *symbols-to-export*
-                  *export-class-name-p*)
-              `(progn
-                 ,result
-                 (eval-when (:compile-toplevel :load-toplevel :execute)
-                   ;; Don't try to export symbols that don't belong to *package*.
-                   ;; This can happen when inheriting from a class and
-                   ;; overriding some slot.
-                   (export '(,@(remove-if (lambda (sym)
-                                            (not (eq (symbol-package sym) *package*)))
-                                (append (when *export-class-name-p*
-                                          (list name))
-                                 *symbols-to-export*)))
-                           ,(package-name *package*)))
-                 (find-class ',name nil))
-              result))))))
+          `(progn
+             ,@(when *predicate-name-transformer*
+                 (let ((pred-name (funcall *predicate-name-transformer* name)))
+                   `((defun ,pred-name (object)
+                       (subtypep (type-of object) ',name))
```

Comment:
subtypep is usually much slower than typep (and it has no correctness guarantees either). is there any reason this is not simply

```
(typep object ',name)
```

Reply:
Yes, the reason is that this more "do what I mean" and in line with CL idioms: in Common Lisp, a predicate returns T on subtypes, so I'm trying to follow the same rule here.

Reply:
maybe i'm too slow, but aren't these two equivalent in semantics (but not in speed):

1. `(typep x 'mytype)`
2. `(subtype (type-of x) 'mytype)`

`typep` also considers subtyping, doesn't it? or did my CL knowledge rot this much in the past few years?

do you have an example?

Reply:
You are right, as I mentioned elsewhere, this was me misunderstanding what `typep` does.  I've fixed it.